### ISSUE TITLE
Changing slug does not update references to entry

### ISSUE THREAD

**Issue by Katie-Adams** [storyblok/storyblok#775]
**The issue occurs while working with:** (check one with "x")   
- [x] *app.storyblok.com (CMS - Interface)*   
- [x] *api.storyblok.com (CMS - Content Delivery API)*
- [ ] *mapi.storyblok.com (CMS - Management API)*
- [ ] *capi.storyblok.com (Commerce - API)*     
- [ ] *Commerce - Interface*   
- [ ] *Other:* 

**Current behavior:**

A pre-existing entry, i.e. a news article, needs its slug updating so I open the Settings panel and change it. I navigate to a different page where a reference/link to the update entry is. On the CMS, it appears to have updated and is using the new slug. However, when pulling the content from the links API, we are still served the old URL. To fix this, I have to go through every single reference to the entry, remove the link, and then readd it. This is made harder once I've already changed the slug as I can't filter by the old URL. 

**Expected behavior:**

When I update a slug, I would expect everywhere that it is referenced to be updated as well.

**Comment by demirgazetic** [comment]
Hello @Katie-Adams, something to check out and maybe helps you fix your issue:
https://www.storyblok.com/cl/url-resolving

**Comment by mariuszmodzelewski** [comment]
@demirgazetic Is there a change planned for cached_url to display the updated url?

**Comment by rasco** [comment]
> When I update a slug, I would expect everywhere that it is referenced to be updated as well.

I agree! Or at least give editors an ability to update all references

**Comment by Tasin5541** [comment]
Facing the same issue. We now have to update tons of references because of it.

**Comment by Katie-Adams** [comment]
One of the workarounds we're considering is an automated job that runs when a slug is updated to check for existing references and updates them to the new one. However it does feel like this is something that the CMS should do out of the box, same as it flags references to an asset when you try to delete it.

**Comment by WL1981** [comment]
Same issue here!

When changing an entry’s slug in Storyblok, the new slug is reflected in the CMS, but references still point to the old slug. This causes outdated links in the API response, requiring manual updates.

Expected behavior: References should update automatically when a slug changes to avoid broken or outdated links.

This is quite frustrating and makes content management inefficient. Is there any update or workaround for this?

**Comment by AdrianFahrbach** [comment]
It seems like this also affects fields where the link has been set and then removed again.

Our solution to this is `resolve_links: 'url'`. When this is set and the link is a story, then you get a small story object like this:

```
{
  "cached_url": "/en/this/is/wrong/foo-bar",
  "fieldtype": "multilink",
  "id": "ab0e4d24-1d7d-4e5f-a5b3-b6l5ra2de6e5",
  "linktype": "story",
  "prep": true,
  "story": {
    "full_slug": "en/this/is/correct/foo-bar",
    "id": 294810,
    "name": "Foo bar",
    "slug": "foo-bar",
    "url": "this/is/correct/foo-bar",
    "uuid": "fb1e2d75-3d7d-4e5f-c5f3-b6f3ca2de4e5"
  },
  "url": ""
}
```

You can then use `story.full_slug`, but you have to keep in mind that it is missing the leading slash.
I'm also not quite sure while the uuid is called id in the link and uuid in the story.

@alvarosabu, @christianzoppi, @manuelschroederdev, @iamfortune, @samuells Is the Storyblok SDK/API still maintained? This issue is a quite crucial error for a CMS and has been open for 3 years without a response. The last closed issue was 2 years ago. Is there a better way to report issues than through GitHub?

**Comment by alvarosabu** [comment]
Hi @Katie-Adams, we are closing this issue because this repository was used in the early days for support-related questions, but we've since transitioned to more effective channels for helping our users.

Please, refer to the following resources:

Storyblok Github Org: https://github.com/storyblok
Storyblok Docs: https://www.storyblok.com/docs
Official Support: https://support.storyblok.com
Join our Discord: https://www.storyblok.com/join-discord

Thanks for the understanding 🙏🏻
